File: programmers/Level1/CPP/제일 작은 수 제거하기.cpp
#include <vector>
using namespace std;

vector<int> solution(vector<int> arr) {
    vector<int> answer;
    
    if(arr.size() == 1){
        answer.push_back(-1);
    }
    else{
        int min = 0;
        for(int i = 0; i < arr.size(); i++){
            if(arr[min] > arr[i]) min = i;
        }
        
        for(int i = 0; i < arr.size(); i++){
            if(min == i) {
                continue;
            }
            else{
                 answer.push_back(arr[i]);
            }
        }
    }
    
    return answer;
}

File: programmers/Level1/JAVA/제일 작은 수 제거하기.java
class Solution {
    public int[] solution(int[] arr) {
        int[] answer = {};
        
        if(arr.length == 1){
            answer = new int[1];
            answer[0] = -1;
        }
        else{
            int min = 0;
            for(int i = 1; i<arr.length; i++){
                if(arr[min] > arr[i]) min = i;
            }
            answer = new int[arr.length-1];
            int j = 0;
            for(int i = 0; i<arr.length; i++){
                if(arr[min] == arr[i]) continue;
                else {answer[j] = arr[i]; j++;}
            }
            
        }
        
        return answer;
    }
}
